mlrun/api/crud/model_endpoints.py

```diff
@@ -1034,45 +1182,254 @@ def build_kv_cursor_filter_expression(
+
+        :param endpoint_id:      The unique id of the model endpoint.
+        :param metrics:          A list of metrics to return for the model endpoint.
+        :param start:            The start time of the metrics.
+        :param end:              The end time of the metrics.
+
+        :return: A dictionary of metrics in which the key is a metric name and the value is a Metric object that also
+                 includes the relevant timestamp. More details about the Metric object can be found under
+                 mlrun.api.schemas.Metric.
+        """
+
+        if not metrics:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Metric names must be provided"
+            )
+
+        # Initialize metrics mapping dictionary
+        metrics_mapping = {}
+
+        # Getting the path for the time series DB
+        events_path = (
+            mlrun.mlconf.model_endpoint_monitoring.store_prefixes.default.format(
+                project=self.project,
+                kind=mlrun.api.schemas.ModelMonitoringStoreKinds.EVENTS,
             )
         )
+        (
+            _,
+            _,
+            events_path,
+        ) = mlrun.utils.model_monitoring.parse_model_endpoint_store_prefix(events_path)
 
-    @staticmethod
-    def _convert_to_cron_string(cron_trigger):
-        """Converting the batch interval dictionary into a ScheduleCronTrigger expression"""
-        return "{} {} {} * *".format(
-            cron_trigger["minute"], cron_trigger["hour"], cron_trigger["day"]
-        ).replace("None", "*")
+        # Retrieve the raw data from the time series DB based on the provided metrics and time ranges
+        frames_client = mlrun.utils.v3io_clients.get_frames_client(
+            token=self.access_key,
+            address=mlrun.mlconf.v3io_framesd,
+            container=self.container,
+        )
+
+        try:
+            data = frames_client.read(
+                backend="tsdb",
+                table=events_path,
+                columns=["endpoint_id", *metrics],
+                filter=f"endpoint_id=='{endpoint_id}'",
+                start=start,
+                end=end,
+            )
+
+            # Fill the metrics mapping dictionary with the metric name and values
+            data_dict = data.to_dict()
+            for metric in metrics:
+                metric_data = data_dict.get(metric)
+                if metric_data is None:
+                    continue
+
+                values = [
+                    (str(timestamp), value) for timestamp, value in metric_data.items()
+                ]
+                metrics_mapping[metric] = mlrun.api.schemas.Metric(
+                    name=metric, values=values
+                )
+        except v3io_frames.errors.ReadError:
+            logger.warn("Failed to read tsdb", endpoint=endpoint_id)
+        return metrics_mapping
+
+
+class _ModelEndpointSQLStore(_ModelEndpointStore):
```

Comment:
I feel like the actual implementation of the `ModelEndpointStore` should be in a separate file. As this is only the abstract class yet and there will be more implementation, I think we should separate it before it grows much more.